### PR TITLE
Added logic to truncate AMM price if it is too big compared to 1 ETH

### PR DESCRIPTION
**Pull request by NakedFool**
## What does this PR do and why?
Fixes - https://github.com/jbx-protocol/juice-interface/issues/1909

Quick summary of thoughts
- I've initially wanted to expand box, but displaying number this big doesn't really make sense to me 💭 
- I've also wanted to use `text-overflow: ellipsis` but the problem there is that it would put ellipsis on our whole string. 
   for example `190,000,000,000 BARK/ 1 ETH` would become `190,000,000 ...` without user seeing that it is price compared to `1 ETH`
- Created a small function that does what i want, you pass `string` and desired `length` of a string and it will return a truncated string with an ellipsis at the end.

## Screenshots or screen recordings

**Before**
image
**After**
image

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/AMMPrices/TokenAMMPriceRow.tsx [hunk 1]
```diff
@@ -82,7 +88,7 @@ export default function TokenAMMPriceRow({
             overlayInnerStyle={{ ...fontStyle }}
           >
             <ExternalLink href={exchangeLink} style={{ fontWeight: 400 }}>
-              {`${formattedNum(WETHPrice)} ${tokenSymbol}/1 ETH`}
+              {`${preparePrice(WETHPrice)} ${tokenSymbol}/1 ETH`}
```
Comment (johnnyd-eth) [inline src/components/AMMPrices/TokenAMMPriceRow.tsx:91]: **Nit:** think we could just name this `formatPrice` for consistency

File: src/components/AMMPrices/TokenAMMPriceRow.tsx [hunk 2]
```diff
@@ -56,6 +57,12 @@ export default function TokenAMMPriceRow({
     )
   }
 
+  const preparePrice = (price: string) => {
```
Comment (johnnyd-eth) [inline src/components/AMMPrices/TokenAMMPriceRow.tsx:60]: .

File: src/utils/formatString.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const truncateString = (str: string, length = 50) => {
```
Comment (johnnyd-eth) [inline src/utils/formatString.ts:1]: **Praise**

**Comment by johnnyd-eth** [comment]
Great find and pickup here @NakedFool. As you mentioned, I would prefer if we still showed an accurate value here rather than ellipses. WDYT about making a `scientificNotation` function so we would show `1.966424e8` (to appropriate preciesion) or something rather than just cutting it off?

**Review by johnnyd-eth — CHANGES_REQUESTED** [review]
.

File: src/components/AMMPrices/TokenAMMPriceRow.tsx (hunk 1 above)
Comment (NakedFool) [reply src/components/AMMPrices/TokenAMMPriceRow.tsx:91]: agreed!

**Comment by NakedFool** [comment]
> Great find and pickup here @NakedFool. As you mentioned, I would prefer if we still showed an accurate value here rather than ellipses. WDYT about making a `scientificNotation` function so we would show `1.966424e8` (to appropriate precision) or something rather than just cutting it off?

Agreed, yeah it feels like this can be more tightened, i'm gonna work on it! <3

**Comment by johnnyd-eth** [comment]
Sounds good! Let me know when this is ready for another look @NakedFool

**Comment by NakedFool** [comment]
> Sounds good! Let me know when this is ready for another look @NakedFool

heyo! I've updated branch with the logic you suggested, lemme quick explain what is going on now there.

As you said we want to display correct number, so therefore i saw that in `utils` there is already function named `truncateLongNumber`

But we do not want to always truncate number as in most cases it won't be that long number, therefore i created utility func `formatOrTruncate` that is either formatting or truncating number based on `limit` that is passed as arg in that func.

I've set limit to be trillion, as numbers above are too big for box and numbers below it fits well, let me know what you think of this approach! 🍻

File: src/components/AMMPrices/TokenAMMPriceRow.tsx [hunk 4]
```diff
@@ -56,6 +56,14 @@ export default function TokenAMMPriceRow({
     )
   }
 
+  const formatPrice = (price: string) => {
+    const p = parseInt(price, 10)
+    const formatLimit = 1e12 - 1 // trillion
```
Comment (johnnyd-eth) [inline src/components/AMMPrices/TokenAMMPriceRow.tsx:61]: **Suggestion: ** add `const` for trillion in `constants/numbers` and use it here :)

**Comment by johnnyd-eth** [comment]
Screen Shot 2022-09-07 at 10 47 14 am

Not sure what's going on with `truncateLongNumber` but this still isn't what we want to show. No idea what the value pictured means. 

I think, since it's potentially so big, we need to show scientific notation when it's above a certain amount. So `1.9735439e12` (however many decimals you think look reasonable).

File: src/components/AMMPrices/TokenAMMPriceRow.tsx (hunk 4 above)
Comment (NakedFool) [reply src/components/AMMPrices/TokenAMMPriceRow.tsx]: Thank you for this suggestion ser, i felt a bit dirty leaving it here :/

**Comment by NakedFool** [comment]
> Screen Shot 2022-09-07 at 10 47 14 am
> 
> Not sure what's going on with `truncateLongNumber` but this still isn't what we want to show. Not sure what the value pictured means.
> 
> I think, since it's potentially so big, we need to show scientific notation when it's above a certain amount. So `1.9735439e12` (however many decimals you think look reasonable).

You are right, i've used `toExponential` instead, lemme know what do you think about latest approach :)

**Comment by johnnyd-eth** [comment]
Screen Shot 2022-09-08 at 1 42 43 pm

Sweet @NakedFool that LGTM 🥇

**Review by johnnyd-eth — APPROVED** [review]

**Pull request merged**